src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,39 +1,76 @@
-import React from 'react';
+import { FC, useEffect, useState } from 'react';
+
+import { User } from '../../types/User';
+
+import { getUser } from '../../api';
+import { useAppDispatch, useAppSelector } from '../../app/hooks';
+import {
+  clearCurrentTodo,
+  selectCurrentTodo,
+} from '../../features/currentTodo';
+
 import { Loader } from '../Loader';
 
-export const TodoModal: React.FC = () => {
+export const TodoModal: FC = () => {
+  const [isLoading, setIsLoading] = useState(false);
+  const [user, setUser] = useState<null | User>(null);
+  const currentTodo = useAppSelector(selectCurrentTodo);
+  const dispatch = useAppDispatch();
+
+  const handleCloseModal = () => {
+    dispatch(clearCurrentTodo());
+  };
+
+  useEffect(() => {
+    setIsLoading(true);
+
+    if (currentTodo) {
+      getUser(currentTodo.userId)
+        .then(setUser)
+        .finally(() => {
+          setIsLoading(false);
+        });
+    }
+  }, [currentTodo]);
```

Comment:
The `useEffect` hook depends on `currentTodo`. Ensure that `getUser` and `setUser` do not change between renders, or consider adding them to the dependency array if they do. This ensures that the effect runs correctly when those dependencies change.